tapdb/asset_minting.go

```diff
@@ -939,14 +939,27 @@ func (a *AssetMintingStore) AddSproutsToBatch(ctx context.Context,
 	// assets committed to within the root commitment specified.
 	assets := assetRoot.CommittedAssets()
 
+	// Before we store any assets from the batch, we need to sort the assets
+	// so that we insert group anchors before reissunces. This is required
+	// to store the asset genesis as the group anchor. All future group
+	// anchor verification depends on inserting group anchors before
+	// reissuances here. We use the raw group anchor verifier since there
+	// is not yet any stored asset group to reference in the verifier.
+	anchorVerifier := tapgarden.GenRawGroupAnchorVerifier(ctx)
```

Comment:
Hmm couldn't we just rely on the upsert behavior here for the genesis?

Reply:
No -  the problem is that from the P.O.V. of the DB queries, the group anchor is defined as the grouped asset with a witness that was stored first.

This wasn't a problem before as only one group member, the group anchor, had a witness.

Now that all group members have a witness, their insertion must be ordered and the anchor must be inserted first. Otherwise the wrong genesis is returned when querying for the group info later on.